### PR TITLE
TASK || Left Side Back End Menu Design fix #18687

### LINKED ISSUE #18687: Left Side Back End Menu Design fix
I have installed the Magento 2.2.5.

I have installed  many extensions which is creating User Menu in the Marketing tab.

Here is the issue that if there are more then 5 columns in the admin side sub menu then last column menu items cut off.
http://prntscr.com/l7jvlc

There should be automatic adjustment menu items so It will not cut off.

### Preconditions (*)
Left Side Back End Menu Design fix

**Information on your environment**
I have installed the Magento 2.2.5.

### Steps to reproduce (*)
Install the multiple extension which is creating sub menu in Marketing tab, and then access the admin side to see how it look like
http://prntscr.com/l7jvlc

### Expected result (*)
There should be automatic adjustment menu items so It will not cut off.

### Actual result (*)
http://prntscr.com/l7jvlc

### PR DESCRIPTION
**Pull request by angelgonzalezi4**
Is a fix to the task Left Side Back End Menu Design fix #18687
URL: https://github.com/magento/magento2/issues/18687

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by kalpmehta — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @kalpmehta, thank you for the review.
[ENGCOM-6187](https://jira.corp.magento.com/browse/ENGCOM-6187) has been created to process this Pull Request

**Comment by kalpmehta** [comment]
Thanks @angelgonzalezi4 for your contribution. I have tested this on one of the website I manage which had this issue for long time, it works like a charm!

File: app/design/adminhtml/Magento/backend/Magento_Backend/web/css/source/module/_menu.less [hunk 1]
```diff
@@ -266,6 +266,22 @@
             .ie11 & {
                 height: 100%;
             }
+           & > ul[role="menu"] {
+                max-width:  ~'calc(@{menu__width} - 100vw)';
+                width: 100vw;
+                display: -webkit-inline-flex;
+                display: -ms-inline-flexbox;
+                display: inline-flex;
```
Comment (dmytro-ch) [inline app/design/adminhtml/Magento/backend/Magento_Backend/web/css/source/module/_menu.less:274]: I guess It's better to use the corresponding mixin:
```suggestion
                .lib-vendor-prefix-display(inline-flex);
```

File: app/design/adminhtml/Magento/backend/Magento_Backend/web/css/source/module/_menu.less [hunk 2]
```diff
@@ -266,6 +266,22 @@
...
+                -webkit-flex-wrap: wrap;
+                -moz-flex-wrap: wrap;
+                -ms-flex-wrap: wrap;
+                -o-flex-wrap: wrap;
+                flex-wrap: wrap;
```
Comment (dmytro-ch) [inline app/design/adminhtml/Magento/backend/Magento_Backend/web/css/source/module/_menu.less:279]:
```suggestion
                .lib-vendor-prefix-flex-wrap(wrap);
```

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
Hello @angelgonzalezi4,
could you please also fix the code styling issues reported by Static Test?
Thank you!

**Comment by kalpmehta** [comment]
Hi @angelgonzalezi4, can you please make the changes suggested by @dmytro-ch ? I would love to see this getting merged.

**Comment by angelgonzalezi4** [comment]
Until the weekend most probably i cant do It bit i Will try it :S

**Comment by VladimirZaets** [comment]
@angelgonzalezi4 , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for collaboration

**Pull request closed without merging**